x-pack/plugins/apm/server/lib/traces/get_trace.ts

```diff
@@ -10,37 +10,109 @@ import { Span } from '../../../typings/es_schemas/Span';
 import { Transaction } from '../../../typings/es_schemas/Transaction';
 import { Setup } from '../helpers/setup_request';
 
-export type TraceAPIResponse = Array<Transaction | Span>;
+export interface TraceAPIResponse {
+  trace: Array<Transaction | Span>;
+  errorsPerTransaction: {
+    [transactionId: string]: number;
+  };
+}
+
+interface TraceErrorsAggBucket {
+  key: string;
+  errors: {
+    doc_count: number;
+  };
+}
+
+interface TraceErrorsAggResponse {
+  transactions: {
+    buckets: TraceErrorsAggBucket[];
+  };
+}
 
 export async function getTrace(
   traceId: string,
   setup: Setup
 ): Promise<TraceAPIResponse> {
   const { start, end, client, config } = setup;
 
-  const params: SearchParams = {
-    index: config.get('apm_oss.transactionIndices'),
+  const query = {
+    bool: {
+      filter: [
+        { term: { [TRACE_ID]: traceId } },
+        {
+          range: {
+            '@timestamp': {
+              gte: start,
+              lte: end,
+              format: 'epoch_millis'
+            }
+          }
+        }
+      ]
+    }
+  };
+
+  const traceParams: SearchParams = {
+    index: [
+      config.get('apm_oss.spanIndices'),
+      config.get('apm_oss.transactionIndices')
+    ],
     body: {
       size: 1000,
-      query: {
-        bool: {
-          filter: [
-            { term: { [TRACE_ID]: traceId } },
-            {
-              range: {
-                '@timestamp': {
-                  gte: start,
-                  lte: end,
-                  format: 'epoch_millis'
+      query
+    }
+  };
+
+  const errorsAggParams: SearchParams = {
+    index: [config.get('apm_oss.errorIndices')],
+    body: {
+      size: 0,
+      query,
+      aggs: {
+        transactions: {
+          terms: {
+            field: 'transaction.id'
+          },
+          aggs: {
+            errors: {
+              filter: {
+                term: {
+                  'processor.event': 'error'
                 }
               }
             }
-          ]
+          }
         }
       }
     }
   };
 
-  const resp = await client<Span | Transaction>('search', params);
-  return resp.hits.hits.map(hit => hit._source);
+  return Promise.all([
+    client<Span | Transaction>('search', traceParams), // query for trace items (spans & transaction)
+    client<Span | Transaction, TraceErrorsAggResponse>( // agg for transaction error count
```

Comment:
`Span | Transaction` is wrong, isn't it? Should probably be `never` or something like that.